File: src/main/java/com/nurflugel/dependencyvisualizer/data/pojos/Person.java
package com.nurflugel.dependencyvisualizer.data.pojos;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import java.util.HashSet;
import java.util.Set;

/** A person is a special type of object, with birth dates, death dates, and spouses. */
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@SuppressWarnings("Lombok")
public class Person extends BaseDependencyObject
{
  private Set<String> spouses   = new HashSet<>();
  private String      birthDate;
  private String      deathDate;

  public Person(String name, String type)
  {
    super(name, type);
  }

  public Person(String name, String[] notes, String type)
  {
    super(name, notes, type);
  }

  public void addSpouse(Person spouse)
  {
    spouses.add(spouse.getName());
  }

  @Override
  public String toString()
  {
    return getDisplayName();
  }

  public void removeAllSpouses()
  {
    spouses.clear();
  }
}

File: src/main/kotlin/com/nurflugel/dependencyvisualizer/data/pojos/Person.kt
package com.nurflugel.dependencyvisualizer.data.pojos

/**
 * A person is a special type of object, with birth dates & death dates.
 * Spouses are recorded as a list of Marriages in the dataset.
 */
class Person(name: String, type: String) : BaseDependencyObject(name, type) {
    var birthDate: String? = null
    var deathDate: String? = null

    override fun toString(): String {
        return displayName
    }
}
